src/modules/hooks/useBlockExplorer.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback } from 'react';
+import { useChains } from 'wagmi';
+
+type ChainEntityType = 'address' | 'tx' | 'token';
+
+interface IChainEntity {
+    type: ChainEntityType;
+    chainId?: number;
+    id: string;
+}
+
+export const useBlockExplorer = () => {
+    const chains = useChains();
+
+    const getChainEntityUrl = useCallback(
+        ({ type, chainId, id }: IChainEntity) => {
+            const chain = chainId ? chains.find((chain) => chain.id === chainId) : chains[0];
+            if (!chain) {
+                throw new Error(`useBlockExplorer: Chain with id ${chainId} not found`);
+            }
```

Comment:
Given that we default it to `chains[0]`, is this check needed?

Reply:
✅ https://github.com/aragon/ods/pull/220/commits/c196c4e5dc88c8714b7dddc28bbd80614dc4652e